include/mc_rtc/gui/Polyhedron.h

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2015-2019 CNRS-UM LIRMM, CNRS-AIST JRL
+ */
+
+#pragma once
+
+#include <mc_rtc/gui/details/traits.h>
+#include <mc_rtc/gui/elements.h>
+#include <mc_rtc/gui/types.h>
+
+namespace mc_rtc
+{
+
+namespace gui
+{
+
+/** Polyhedron should display a polyhedron or a set of polyhedrons
+ *
+ * A color is provided to display the polyhedron(s)
+ *
+ * \tparam GetT Should return an std::vector<Eigen::Vector3d> (one polyhedron) or an
+ * std::vector<std::vector<Eigen:Vector3d>> (list of polyhedrons)
+ *
+ */
+template<typename GetVerticesT>
+struct PolyhedronImpl : public Element
+{
+  static constexpr auto type = Elements::Polyhedron;
+
+  PolyhedronImpl(const std::string & name, const PolyhedronConfig & config, GetVerticesT get_vertices_fn)
+  : Element(name), config_(config), get_vertices_fn_(get_vertices_fn)
+  {
+    static_assert(details::CheckReturnType<GetVerticesT, PolyhedronTriangles>::value,
+                  "Polyhedron vertices callback must return an std::vector<Eigen::Vector3d> (triangle set with points "
+                  "ordered clockwise)");
+  }
+
+  /** Invalid element */
+  PolyhedronImpl() {}
+
+  static constexpr size_t write_size()
+  {
+    return Element::write_size() + 2 + PolyhedronConfig::write_size();
+  }
+
+  void write(mc_rtc::MessagePackBuilder & builder)
+  {
+    builder.write(get_vertices_fn_());
+    builder.write(PolyhedronColors{});
```

Comment:
Just add a null entry:
```suggestion
    builder.write();
```

Reply:
Thanks, I hadn't paid attention we could do that :)